spark/v3.3/build.gradle

```diff
@@ -59,6 +59,10 @@ project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}") {
     implementation project(':iceberg-parquet')
     implementation project(':iceberg-arrow')
     implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}:${libs.versions.scala.collection.compat.get()}")
+    if (scalaVersion == '2.12') {
+      // scala-collection-compat_2.12 pulls scala 2.12.17 and we need 2.12.18 for JDK 21 support
```

Comment:
does this also apply for Scala 2.13?

Reply:
apparently not, i think the Ci would detect that

Reply:
do you remember what the exact error msg was here?

Reply:
i don't, i should have kept a note